built-ins/authoring-built-ins.md

```diff
@@ -0,0 +1,132 @@
+
+* /*{templateSpecId}*/versions/*{versionName}*/**mainTemplate.json**
+
+**If your template spec contains artifacts** you'll need to export the template spec using [Az Powershell](https://docs.microsoft.com/en-us/powershell/azure/new-azureps-module-az) cmdlets. That process is as follows:
+
+* Use `Set-AzContext` to set the subscription where your template spec is located to the active context
+* Run `Export-AzTemplateSpec -ResourceGroupName {rgName} -Name {templateSpecId} -Version {versionName} -OutputFolder {theRootBuiltInsDirectory}/{templateSpecId}/versions/{versionName}`
+* Navigate to the output folder and rename the .json file created there (it'll be named with the format "*{templateSpecId}.{versionName}.json*") to **mainTemplate.json**
+
+### Step 4: Add metadata for your template spec
+
+Each built-in template spec requires additional metadata for the overall template spec, as well as each individual version. We'll start with the overall template spec metadata....
+
+#### Metadata for overall template spec:
+Create a file in your template spec directory with the name 'metadata.json' (eg: * /*{templateSpecId}*/metadata.json) and populate it with the following contents to start things off:
+
+    {
+        "displayName" : "User Friendly Built-in Name",
+        "description": "Very short description of the built-in's purpose",
+        "documentationUrl": "https://www.microsoft.com/myDocumentation",
+        "category": "General",
+    }
+
+Replace the example values with values relevant to your built-in. The **required** properties are as follows:
+
+* **displayName**: This will be the user friendly (it can contain spaces) name displayed to customers in the Azure Portal (*currently the built-in UI is pending implementation*). It should be as short as possible and follow normal casing.
+* **description**: Very short description of the built-in's purpose. For example: "Deploys a Windows VM."
+* **category**: The relevant category for your built-in. Currently the following categories are supported (they will be expanded in the near future, but additional requests can be raised under our github issue tracking):
+  * *"General"* - Common catch-all category for general built-ins that do not fit any other available category
+  * *"Quickstart"* - Provides a starting point for a common deployment scenario (deploying a VM for example)
+
+Additionally the following **optional** properties can be set:
+
+* **documentationUrl**: An absolute URL to documentation related to your built-in. Having this property/value present is strongly recommended. For built-ins contributed by non-Microsoft third parties, the URL may be strongly scrutinized for longevity/security reasons. 
```

Comment:
Should documentationUrl be required for all built-ins? (If this gets large enough, we may need to monitor these URLs and raise incidents, but definitely not anytime soon.)

Reply:
+1 on this. We should make sure we have a plan in place to avoid any doc links becoming stale over time

Reply:
I'm open to making this required. My hesitation was with the long-term potential of built-ins contributed in the future by third parties. If we open up to external built-in contributors, we run into the staleness issue as well as URL safety, in which case we may not want to have a documentationUrl if the contributor cannot meet our documentation hosting needs (and I don't think we want to take on the responsibility of hosting docs for third parties if we do open up contributions to external parties).

For now, perhaps I'll doc it as required for built-ins authored by Microsoft employees.

Reply:
@majastrz Thoughts?

Reply:
We can start strict by making it required and then relax when we have a need for it (when we open up to 3rd parties).